Add documentation links

piyushpalawat99 opened the pull request:
Documentation links have been added to WindowExtension.cpp and SceneExtension.cpp

piyushpalawat99 commented:
The SceneExtension.cpp file contains the Random expressions which have already been discussed in the [Expressions](http://wiki.compilgames.net/doku.php/gdevelop5/all-features/expressions) page. It also contains the string and number comparison conditions which I believe, do not belong to the category.

Should these functions be added in some other file?

4ian commented:
Ideally they should be in another file. In practice, don't bother moving them and let them here.
To be sure that the help page is pointing to the proper location, use `SetHelpPath`:

```
extension
      .AddAction(...)
      .SetHelpPath("/somewhere").
      .AddParameter(...)
      .AddParameter(...)
      ...
```

piyushpalawat99 commented:
For now, I have added them to the advanced conditions page. I don't think there is some other page that suits them more currently (mainly because they belong to the "Other" category).

4ian commented:
Yeah advanced is fine, they should actually not be used a lot except in some "advanced" situations :)

piyushpalawat99 commented:
I'll be working on AdvancedCondition.cpp file soon, so I request you to keep the merging process on hold.

The reason behind not working on it yet was confusion regarding the path. Thanks for clearing it.

piyushpalawat99 commented:
Done:)
Had to make a new page and add link in the functions page.

4ian commented on Core/GDCore/Extensions/Builtin/AdvancedExtension.cpp:
```diff
@@ -18,7 +18,7 @@ void GD_CORE_API BuiltinExtensionsImplementer::ImplementsAdvancedExtension(
           "Florian Rival",
           "Open source (MIT License)")
-      .SetExtensionHelpPath("" /*TODO: Add a documentation page for this */);
+      .SetExtensionHelpPath("/events/functions/return");
```
I would prefer if you can use SetHelpPath on the 3 actions. The extension itself is a mix of things, but should not have a help page pointing to something that is specific like this to only a few actions.
When done, you can actually remove `.SetExtensionHelpPath("" /*TODO: Add a documentation page for this */)` entirely :)

4ian commented on Core/GDCore/Extensions/Builtin/SceneExtension.cpp:
```diff
@@ -18,7 +18,7 @@ void GD_CORE_API BuiltinExtensionsImplementer::ImplementsSceneExtension(
           "Florian Rival",
           "Open source (MIT License)")
-      .SetExtensionHelpPath("" /*TODO: Add a documentation page for this */);
+      .SetExtensionHelpPath("/interface/scene-editor/events");
```
Same here, be careful because doing this you'll be adding a help page for expressions like Random (they might not be shown in the editor, but one day they will, and in this case we would have wrong link for RandomXXX expressions, which is worse than no link).
Instead, use SetHelpPath to set the help paths for each condition/action.

4ian commented on Core/GDCore/Extensions/Builtin/WindowExtension.cpp:
```diff
@@ -18,7 +18,7 @@ void GD_CORE_API BuiltinExtensionsImplementer::ImplementsWindowExtension(
                                "Florian Rival",
                                "Open source (MIT License)")
-      .SetExtensionHelpPath("" /*TODO: Add a documentation page for this */);
+      .SetExtensionHelpPath("/all-features/window");
```
This looks good 👍

4ian commented on Core/GDCore/Extensions/Builtin/SceneExtension.cpp:
```diff
@@ -170,6 +170,7 @@ void GD_CORE_API BuiltinExtensionsImplementer::ImplementsSceneExtension(
                     _("Other"),
                     "res/conditions/egal24.png",
                     "res/conditions/egal.png")
+      .SetHelpPath("/all-features/advanced-conditions")
```
👍

piyushpalawat99 commented:
I have added the links for individual actions/conditions. I have also added links individually for the random expressions as it might save some work in the future :)

piyushpalawat99 commented:
Why are there so many help links in [BaseObjectExtension.cpp](https://github.com/4ian/GDevelop/blob/master/Core/GDCore/Extensions/Builtin/BaseObjectExtension.cpp) that guide to the base object page even though they are not explained there?

I'd rather suggest adding pages linked to the base object page. I can do it but it might take time so a bit more hold for this pr :')

4ian commented:
> Why are there so many help links in BaseObjectExtension.cpp that guide to the base object page even though they are not explained there?

Probably this can be improved :)

>  I can do it but it might take time so a bit more hold for this pr :')

If you prefer, we can go ahead with this one and then make another one for other links :)

4ian commented:
Beware, you had several changes that were syntax invalid (extra ;, missing parenthesis). Also I see that you're working on master branch. I would suggest for future modifications to see how to do branches with git and work on a branch. Otherwise you'll have difficulties pulling my latest changes from master - especially because I "squash" all your commits inside one - which can confuse git.

piyushpalawat99 commented:
> If you prefer, we can go ahead with this one and then make another one for other links :)

All the conditions/actions in the file belong to the "Common conditions/actions of an object" category. Although many of them have already been explained, I will add a few sections in the base object page to explain the unexplained features rather than making a new page so there won't be a need for a new pull request.

> Beware, you had several changes that were syntax invalid (extra ;, missing parenthesis).

Sorry for the inconvenience caused. I'll take extra care while doing something similar in the future. Thanks for bearing with the mistakes and fixing them.

> I would suggest for future modifications to see how to do branches with git and work on a branch.

I just saw how to work with branches in git. Sure, all the future changes will be done in a different branch as it also allows us to easily go to previous commits :)

Thanks for the help and you can now merge this pull request because as mentioned above, there's no need to do changes to the file, I'd rather be doing changes in the wiki page :)
